editor/src/components/canvas/canvas-strategies/draw-to-insert-strategy.spec.browser2.tsx

```diff
@@ -458,6 +459,71 @@ describe('Inserting into absolute', () => {
       `),
     )
   })
+
+  it('when drag ends outside the canvas in insert mode, it is cancelled', async () => {
+    const renderResult = await renderTestEditorWithCode(inputCode, 'await-first-dom-report')
+
+    const targetElement = renderResult.renderedDOM.getByTestId('bbb')
+    const targetElementBounds = targetElement.getBoundingClientRect()
+    const canvasControlsLayer = renderResult.renderedDOM.getByTestId(CanvasControlsContainerID)
+
+    const startPoint = slightlyOffsetWindowPointBecauseVeryWeirdIssue({
+      x: targetElementBounds.x + 5,
+      y: targetElementBounds.y + 5,
+    })
+
+    const endPoint = slightlyOffsetWindowPointBecauseVeryWeirdIssue({
+      x: targetElementBounds.x + 1005,
+      y: targetElementBounds.y + 1005,
+    })
+
+    mouseMoveToPoint(canvasControlsLayer, startPoint)
+
+    await startInsertMode(renderResult.dispatch)
+
+    mouseDragFromPointToPointNoMouseDown(canvasControlsLayer, startPoint, endPoint)
```

Comment:
I'm confused - this wouldn't trigger an insertion anyway if there were no mousedown event. It seems more than likely that you should be triggering a full drag here. If you need the mouse events to fire at different DOM elements then it might make sense to call the `mouseDownAtPoint`, `mouseMoveToPoint` and `mouseUpAtPoint` events separately so that you can do that.

Reply:
does this test break (as it should) on master?

Reply:
`Should drag to insert into targets smaller than the element` does

Reply:
so my understanding is that when I'm inserting an element via the insert menu, when I `mousedown` on an insert menu item, the insertion is initiated by the mousedown event handler,  and I wanted to model the same process. 

Is it better approach to 
- enter insert mode
- fire a mousedown event on the insert menu
- move the cursor to the canvas
- move the cursor somewhere far out
- fire mouseup and check whether the interaction was cancelled?
?

Reply:
OK I got it, I modified the test so that it break on master but succeeds on this branch, with all the required mouse events